src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -38,6 +49,38 @@ internal class MissingSetupComponentRegistrationService : IMissingSetupComponent
         private IVsSolution? _vsSolution;
         private bool? _isVSFromPreviewChannel;
 
+        private readonly object _lock = new();
+        private static HashSet<string>? _netCoreRegistryKeyValues = null;
+
+        private HashSet<string> NetCoreRegistryKeyValues
```

Comment:
I would move all the registry access to another type that implements an `IRegistry` interface. I _removed_ a service like that in https://github.com/dotnet/project-system/pull/7880 because it was no longer needed, but you can bring it back.

That separates one concern (accessing the registry) from the work of this type (checking if the necessary runtime is installed) and will make this easier to understand and test with unit tests.